src/views/Home.jsx

```diff
@@ -1,38 +1,34 @@
 import './Home.css';
 import { SingleList } from '../components';
-import { useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import { createList } from '../api';
+import TextInputElement from '../components/TextInputElement';
 
 export function Home({ data, setListPath, userId, userEmail }) {
-	const [listName, setListName] = useState('');
 	const navigate = useNavigate();
 
-	const handleChange = (event) => {
-		setListName(event.target.value);
-	};
-
 	const handleSubmit = async (event) => {
 		event.preventDefault();
+		const listName = event.target['list-name'].value;
 		const currentLists = data.map((list) => {
 			return list.name.toLowerCase();
 		});
 
-		if (currentLists.includes(listName.toLowerCase())) {
-			alert('The list already exists. Please enter a different name.');
-			setListName('');
-			return;
-		}
-
 		try {
+			if (currentLists.includes(listName.toLowerCase())) {
+				alert('The list already exists. Please enter a different name.');
+				return;
+			}
+
 			const listPath = await createList(userId, userEmail, listName);
```

Comment:
move if block into the try block - this allows to implement form reset in finally block